File: app/src/main/java/com/toandoan/todoapp/screen/main/MainActivity.java
package com.toandoan.todoapp.screen.main;

import android.arch.lifecycle.LifecycleOwner;
import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.toandoan.todoapp.R;
import com.toandoan.todoapp.databinding.ActivityMainBinding;
import com.toandoan.todoapp.di.AppModule;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity implements LifecycleOwner {
    @Inject
    MainViewModel mViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        DaggerMainComponent.builder()
                .mainModule(new MainModule(this, this))
                .appModule(new AppModule())
                .build()
                .inject(this);

        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setViewModel(mViewModel);
        mViewModel.onCreate();

        getSupportActionBar().setTitle(R.string.title_main_screen);
    }

    @Override
    protected void onStop() {
        mViewModel.onStop();
        super.onStop();
    }
}

File: app/src/main/java/com/toandoan/todo/screen/main/MainActivity.kt
package com.toandoan.todo.screen.main

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import com.toandoan.todo.R
import com.toandoan.todo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

  var viewModel: MainViewModel? = null

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    viewModel = MainViewModel(this)
    var binding: ActivityMainBinding = DataBindingUtil.setContentView(this,
        R.layout.activity_main)
    setSupportActionBar(binding.toolbar)

    binding.viewModel = viewModel
  }

  override fun onStart() {
    super.onStart()
    viewModel?.onStart()
  }

  override fun onCreateOptionsMenu(menu: Menu?): Boolean {
    menuInflater.inflate(R.menu.main_menu, menu)
    return super.onCreateOptionsMenu(menu)
  }

  override fun onOptionsItemSelected(item: MenuItem?): Boolean {
    when (item!!.itemId) {
      android.R.id.home -> onBackPressed()
      R.id.menu_delete -> {
        viewModel?.onDeleteAllTaskClicked()
      }
    }
    return super.onOptionsItemSelected(item)
  }

  override fun onStop() {
    viewModel?.onStop()
    super.onStop()
  }

}
